src/main/java/bio/terra/workspace/service/get/GetService.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.workspace.service.get;
+
+import bio.terra.workspace.app.configuration.ApplicationConfiguration;
+import bio.terra.workspace.common.exception.SamApiException;
+import bio.terra.workspace.common.utils.SamUtils;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.generated.model.GetWorkspaceRequestBody;
+import bio.terra.workspace.generated.model.WorkspaceDescription;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GetService {
+  private ApplicationConfiguration configuration;
+  private final WorkspaceDao workspaceDao;
+  private final SamService samService;
+
+  @Autowired
+  public GetService(
+      ApplicationConfiguration configuration, WorkspaceDao workspaceDao, SamService samService) {
+    this.configuration = configuration;
+    this.workspaceDao = workspaceDao;
+    this.samService = samService;
+  }
+
+  public WorkspaceDescription getWorkspace(
+      GetWorkspaceRequestBody body, AuthenticatedUserRequest userReq) {
+
+    try {
+      samService.isAuthorized(
+          body.getAuthToken(),
```

Comment:
If we decide to put tokens in headers, this should use the token from userReq instead of from the request body